src/core/lib/transport/metadata.h

```diff
@@ -160,4 +160,9 @@ void grpc_mdelem_unref(grpc_mdelem md);
 void grpc_mdctx_global_init(void);
 void grpc_mdctx_global_shutdown();
 
+inline bool grpc_mdelem_eq_static(grpc_mdelem a_static, grpc_mdelem b_static) {
```

Comment:
Please move this up to line 128, next to `grpc_mdelem_eq()`, so that people looking at this file can tell that there are two alternatives.

Also, please document how this function differs from `grpc_mdelem_eq()` (it only works on static metadata and only checks the value, not the key), so that it's clear to readers when to use one vs. the other.

I suggest renaming this function to something like `grpc_mdelem_static_value_eq()` to make the behavior a bit more clear.

Reply:
The reason it is down here is because of the macros it relies on being defined after line 128. Should I forward declare up there, or move things around further?

I will add a comment and change the function name.

Reply:
I have also shifted the macros up a bit so I can move the function as you stated.

Reply:
Please push your changes before resolving or replying. I can't see the changes here.